### PR TITLE
finish dialog logic including timing, persistence

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: src/js/controllers/DialogController.js [hunk 1]
```diff
@@ -2,23 +2,33 @@
   'use strict';
 
   // Coordinate interactive dialogs.
-  function DialogController($location, $window, $routeParams, dialogs,
-                            responsesCache, uuid, Routes, Dialogs) {
+  function DialogController($location, $window, $routeParams, dialogsService,
+                            responsesCache, uuid, Routes, Dialogs, DIALOGS) {
```
Comment (eschlange) [inline src/js/controllers/DialogController.js]: Should the original "Dialogs" var still be injected here?

Comment (eschlange) [inline src/js/controllers/DialogController.js]: Never mind, it took me awhile to read through everything to understand the significance of the two dialogs ("Dialog" and "DIALOG".)

**Comment by eschlange** [comment]
Looks good, I'll likely use some similar structures in SiS.

File: src/js/controllers/DialogController.js (hunk 1 above)
Comment (ericcf) [inline src/js/controllers/DialogController.js]: Yeah, my naming scheme is terrible here. I need to refactor...

**Pull request merged**